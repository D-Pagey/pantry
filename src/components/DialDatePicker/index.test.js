import React from 'react';
import userEvent from '@testing-library/user-event';
import { addWeeks, addDays, addMonths, addYears, subDays, subMonths, subYears } from 'date-fns';
import { DialDatePicker } from '.';

const props = {
    date: new Date('March 10, 2019 03:24:00'),
    setDate: () => {}
};

describe('DialDatePicker component', () => {
    it('should render', () => {
        const { container } = render(<DialDatePicker {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render label', () => {
        const label = 'Expiry date?';
        const { getByText } = render(<DialDatePicker {...props} label={label} />);
        getByText(label);
    });

    it.each`
        name           | handler      | direction | index
        ${'addDays'}   | ${addDays}   | ${'Up'}   | ${0}
        ${'addMonths'} | ${addMonths} | ${'Up'}   | ${1}
        ${'addYears'}  | ${addYears}  | ${'Up'}   | ${2}
        ${'subDays'}   | ${subDays}   | ${'Down'} | ${0}
        ${'subMonths'} | ${subMonths} | ${'Down'} | ${1}
        ${'subYears'}  | ${subYears}  | ${'Down'} | ${2}
    `('should call $handler with right value', ({ handler, direction, index }) => {
        const setDate = jest.fn();
        const { queryAllByText } = render(<DialDatePicker {...props} setDate={setDate} />);
        const button = queryAllByText(direction)[index];
        const futureDate = handler(props.date, 1);

        userEvent.click(button);

        expect(setDate).toHaveBeenCalledWith(futureDate);
    });

    it('should handle week increment', () => {
        const setDate = jest.fn();
        const { getByText } = render(<DialDatePicker {...props} setDate={setDate} />);
        const oneWeeksTime = addWeeks(props.date, 1);

        userEvent.click(getByText('Add 1 week'));

        expect(setDate).toHaveBeenCalledWith(oneWeeksTime);
    });
});
