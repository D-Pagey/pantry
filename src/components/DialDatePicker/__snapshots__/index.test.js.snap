// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialDatePicker component should render 1`] = `
.c4 {
  background-color: #288efc;
  border: 0;
  border-radius: 0.25rem;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  height: 45px;
  line-height: 1.2;
  width: 200px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 0 1rem;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(3,max-content);
  grid-column-gap: 1rem;
  grid-row-gap: 1rem;
  margin: 0 0 1rem;
}

.c3 {
  color: blue;
  grid-row: 2;
}

.c2 {
  grid-row: 1;
}

<div
  class="c0"
>
  
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      Up
    </button>
    <span
      class="c3"
      data-testid="dialDatePickerDay"
    >
      27th
    </span>
    <button
      type="button"
    >
      Down
    </button>
    <button
      class="c2"
      type="button"
    >
      Up
    </button>
    <span
      class="c3"
      data-testid="dialDatePickerMonth"
    >
      January
    </span>
    <button
      type="button"
    >
      Down
    </button>
    <button
      class="c2"
      type="button"
    >
      Up
    </button>
    <span
      class="c3"
      data-testid="dialDatePickerYear"
    >
      2020
    </span>
    <button
      type="button"
    >
      Down
    </button>
  </div>
  <button
    class="c4"
    data-testid=""
    type="button"
    width=""
  >
    Add 1 week
  </button>
</div>
`;
